django/db/models/expressions.py

```diff
@@ -412,6 +432,12 @@ def resolve_expression(self, query=None, allow_joins=True, reuse=None, summarize
     def refs_aggregate(self, existing_aggregates):
         return refs_aggregate(self.name.split(LOOKUP_SEP), existing_aggregates)
 
+    def asc(self):
```

Comment:
Well `OrderBy` does not support combinable, and I'd bet there'd be other use-cases where an expression would want to forbid combining itself with another expression. It's not a common use-case though, and makes the API more complicated.

I have no issue renaming CombinableMixin to Combinable. Some time within the next week I plan to go over all the expressions and their docs and do some tidying up. I'll create a commit that shows the Resolvable idea, and we can talk about whether or not it's a good idea then. How's that sound?